src/Stratis.Bitcoin/Consensus/ConsensusManger.cs

```diff
@@ -28,41 +30,55 @@ public class ConsensusManager
 
         private AsyncQueue<(ChainedHeader headerFrom, ChainedHeader headerTo)> toDownloadQueue;
 
-        public ConsensusManager(Network network, ILoggerFactory loggerFactory, IChainedHeaderTree chainedHeaderTree, IChainState chainState, ConsensusSettings consensusSettings, ConcurrentChain concurrentChain, ILookaheadBlockPuller lookaheadBlockPuller)
+        public ConsensusManager(
+            Network network, 
+            ILoggerFactory loggerFactory, 
+            IChainState chainState, 
+            IChainedHeaderValidator chainedHeaderValidator, 
+            ICheckpoints checkpoints, 
+            ConsensusSettings consensusSettings, 
+            ConcurrentChain concurrentChain, 
+            ILookaheadBlockPuller lookaheadBlockPuller,
+            IConsensusRules consensusRules)
         {
             this.network = network;
-            this.chainedHeaderTree = chainedHeaderTree;
             this.chainState = chainState;
             this.consensusSettings = consensusSettings;
             this.concurrentChain = concurrentChain;
             this.lookaheadBlockPuller = lookaheadBlockPuller;
+            this.consensusRules = consensusRules;
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+
+            this.chainedHeaderTree = new ChainedHeaderTree(network, loggerFactory, chainedHeaderValidator, checkpoints, chainState, consensusSettings);
+
+            this.blocksRequested = new Dictionary<uint256, List<Action>>();
```

Comment:
This will not be action, it will be a delegate, but you probably just use it as a placeholder right?

Reply:
Yes when I need it I will work on this 👍